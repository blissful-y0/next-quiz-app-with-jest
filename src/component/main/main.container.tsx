import { useContext } from "react";
import { GlobalContext } from "../../../pages/_app";
import MainUI from "./main.presenter";
import { ChangeEvent } from "react";

function Main() {
  const { setUserName } = useContext(GlobalContext);

  const onChangeInput = (event: ChangeEvent<HTMLInputElement>) => {
    setUserName(event.target.value);
  };

  return <MainUI onChangeInput={onChangeInput} />;
}

export default Main;
