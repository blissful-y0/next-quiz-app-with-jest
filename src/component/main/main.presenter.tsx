import {
  PageWrapper,
  Question,
  StartQuizButton,
  TextArea,
  UserNameInput,
} from "./main.style";
import Link from "next/link";
import { ChangeEvent } from "react";

interface IProps {
  onChangeInput: (event: ChangeEvent<HTMLInputElement>) => void;
}

function MainUI({ onChangeInput }: IProps) {
  return (
    <PageWrapper>
      <Question data-testid="WelcomeToQuiz">
        퀴즈의 세계에 오신 것을 환영합니다!
        <TextArea>당신의 이름을 입력해 주세요...</TextArea>
        <UserNameInput onChange={onChangeInput} />
        <Link href="/quiz">
          <StartQuizButton>퀴즈 시작하기</StartQuizButton>
        </Link>
      </Question>
    </PageWrapper>
  );
}

export default MainUI;
