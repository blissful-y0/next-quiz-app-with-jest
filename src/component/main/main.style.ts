import styled from "@emotion/styled";

export const PageWrapper = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: linear-gradient(
    340deg,
    #009779,
    #00b092,
    #00c9ab,
    #19e2c4,
    #32fbdd
  );
`;

export const Question = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 80%;
  max-width: 1000px;
  height: 70%;
  min-height: 500px;
  color: #009779;
  font-size: 30px;
  font-weight: bold;
  border: 5px solid white;
  border-radius: 25px;
  padding: 1%;
  text-align: center;
  background-color: white;
`;

export const TextArea = styled.div`
  width: 100%;
  font-size: 25px;
  font-weight: normal;
  margin-top: 2%;
  margin-bottom: 2%;
`;

export const UserNameInput = styled.input`
  width: 50%;
  font-size: 40px;
  text-align: center;
  border: 5px solid #009779;
  padding: 2%;
`;

export const StartQuizButton = styled.button`
  margin-top: 3%;
  width: 300px;
  height: 80px;
  background-color: #009779;
  outline: none;
  border: 0px;
  color: white;
  font-size: 40px;
  font-weight: bold;
`;
