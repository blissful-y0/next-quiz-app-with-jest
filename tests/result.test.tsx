import { cleanup, render, screen } from "@testing-library/react";
import Result from "../pages/result";
import { Question, StartQuizButton } from "../src/component/main/main.style";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

describe("Should render the app without crashing", () => {
  it("Renders the page without crash", () => {
    render(<Result />);
  });

  it("renders homepage uncahged", () => {
    const tree = renderer.create(<Result />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(cleanup);
