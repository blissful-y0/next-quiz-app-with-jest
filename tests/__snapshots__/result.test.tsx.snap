// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the app without crashing renders homepage uncahged 1`] = `
<div
  className="css-fsuynv"
>
  <div
    className="css-4udrpq"
  >
    
     님의 결과
    <div
      className="css-1k2igav"
    >
      <div
        className="recharts-responsive-container"
        style={
          Object {
            "height": "100%",
            "maxHeight": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="css-1ar8wk3"
    >
      
       님의 최종 점수는 
       / 10입니다! 문제 풀이에 걸린 총 시간은 
      0초
      입니다!
    </div>
    <h1
      className="css-1ppnsqm"
    >
      오답 노트
    </h1>
    <div
      className="css-1hgjq2u"
    />
  </div>
  <button
    className="css-1pr2grl"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    처음으로
  </button>
</div>
`;
